bin/npg_publish_tree.pl

```diff
@@ -188,46 +188,32 @@ sub handler {
 my @files = grep { -f } $publisher->list_directory($source_directory,
                                                    recurse => 1);
 
-my $mlwh_json_cb = sub {
-  # DataObject, path to file folder, collection file
-  my ($obj, $collection, $file) = @_;
-  if (defined $mlwh_json_filename) { 
+sub write_json {
+  my ($json_filename, $irods_collection) = @_;
+  if (defined $json_filename && (not $json_filename =~ qr/^\s*$/sxm)
+      && defined $irods_collection && (not $irods_collection =~ qr/^\s*$/sxm)) {
```

Comment:
These values come from the CLI and should be checked earlier (there are already some checks). If the values are missing or invalid, there should be an error at that point, with suffiecient information that the user can correct the problem. By the time we reach here, we should be working with valid data.

Reply:
Ok I will remove those two lines